scripts/docker/docker_usage_sum.py

```diff
@@ -0,0 +1,367 @@
+from urllib.error import HTTPError, URLError
+import json
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+
+# A script to collect which dockers are in use and which latest dockers are available
+# Usage: python3 docker_usage_sum.py
+# Output: dockers.in_use.tsv
+# Note: - This script is not perfect. It will not be able to detect dockers that are
+#       imported from other wdl files. It will only detect dockers that are
+#       explicitly defined in the wdl file.
+#       - The script assumes it is executed from the scripts/docker directory, and the
+#       wdl files are in ../../wdl directory.
+#       - The script will retrieve the "latest" tag by date, so if an unofficial tag was
+#       created, after the official tag was created, the script will retrieve the
+#       unofficial tag as the latest tag. (It tries to avoid this by filtering out
+#       tags with no digits).
+#      - The script occasionally uses gcloud to retrieve the latest tag. Its suggested
+#      to have gcloud installed.
+
+# TODO: Future suggestion: have the results be generated for main branch for each merge
+
+def main():
+    current_dir = os.path.abspath(os.path.dirname(__file__))
+
+    print("COLLECTING DOCKERS IN USE...")
+    WDLS_DIR = os.path.abspath(os.path.join(current_dir, "../../wdl"))
+    OUT_SUMMARY_TSV = os.path.join(current_dir, "dockers.in_use.tsv")
+
+    if os.path.exists(OUT_SUMMARY_TSV):
+        os.remove(OUT_SUMMARY_TSV)
+
+    wdl_files = get_wdl_files(dir_to_wdls=WDLS_DIR)
+    global_docker_info = []
+
+    total_files = len(wdl_files)  # Used for Progression calculation
+
+    for index, wdl_path in enumerate(wdl_files, start=1):
+
+        wdl_name = wdl_path
+
+        with open(wdl_path, "r") as file_content:
+            content = file_content.read()
+            pattern = re.compile(r'.*docker:.*"')
+            if pattern.search(content): # If wdl file contains "docker:"
+                matched_lines = []
+                file_content.seek(0)
+                lines = file_content.readlines()
+
+                for line_number, line in enumerate(lines, start=1):
+                    if pattern.search(line):
+                        matched_lines.append((line_number, line.strip()))
+
+                docker_info: list[str] = get_docker_info_from_string(
+                    wdl_lines=matched_lines, wdl_path=wdl_name
+                )
+
+                sorted_info: list = sorted(docker_info, reverse=False)
+
+                global_docker_info.append(sorted_info)
+            else:
+                pass
+
+        # Progression
+        # Calculate the percentage completion
+        progress: float = (index + 1) / total_files * 100
+
+        # Clear the previous line and print the progress
+        print(f"Progress: {progress:.2f}%\r", end="")
+    with open(OUT_SUMMARY_TSV, "a") as tsv_file:
+        # Add header
+        tsv_file.write(f"DOCKER_NAME\tLATEST_TAG\tUSED_TAG\tFILE_LINE\tWDL_PATH")
```

Comment:
should need an `\n` after `WDL_PATH`.

Reply:
Not sure why but adding a `\n` creates an empty line between the header and the first line with the docker-wdl info.